galley/pkg/config/processor/transforms/serviceentry/transformer.go

```diff
@@ -328,6 +329,11 @@ func (t *serviceEntryTransformer) toMcpResource(service *resource.Entry, endpoin
 	entry := &resource.Entry{
 		Metadata: meta,
 		Item:     &se,
+		Origin: &rt.Origin{
+			Name:       service.Metadata.Name,
+			Kind:       "Service",
+			Collection: metadata.K8SCoreV1Services,
+		},
```

Comment:
Rather than creating a new Origin, can we just use the one that belongs to `service`? The resource we're converting from should already have an origin, and I think it makes sense for the synthetic entry to share the same origin.

This would also mean that the transformer doesn't need to worry about the underlying Origin implementation and isn't coupled to kube.

Reply:
done